src/libs/replaygain.liq

```diff
@@ -3,108 +3,100 @@
+    _source = source.replaygain.compute(request.once(_request))
+    source.drop(ratio=ratio, _source)
+    _source.gain()
+  else
+    null()
   end
+end
 
-  s = metadata.map(f, s)
-  amplify(id=id, override="replaygain_track_gain", 1., s)
+# Extract the ReplayGain from the metadata (in dB).
+# @category Metadata
+# @param _metadata Metadata from which the ReplayGain should be extracted.
+def metadata.replaygain(_metadata) =
+  if list.assoc.mem("r128_track_gain", _metadata) then
+    try
+      float_of_string(_metadata["r128_track_gain"]) / 256.
+    catch _ do
+      null()
+    end
+  elsif list.assoc.mem("replaygain_track_gain", _metadata) then
+    replaygain_metadata = _metadata["replaygain_track_gain"]
+    match = r/([+-]?\d*\.?\d*)/.exec(replaygain_metadata)
+    try
+      float_of_string(list.assoc(1, match))
+    catch _ do
+      null()
+    end
+  else
+    null()
+  end
 end
 
-# Compute the ReplayGain for a file (in dB).
+# Get the ReplayGain for a file (in dB).
 # @category File
-# @param ~ratio Decoding ratio. A value of `50` means try to decode the file `50x` faster than real time, if possible. Use this setting to lower CPU peaks when computing replaygain tags.
-# @param fname File name.
-def file.replaygain(~id=null(), ~ratio=50., fname) =
+# @param ~id Force the value of the source ID.
+# @param ~compute Compute ReplayGain if metadata tag is empty.
+# @param ~ratio Decoding ratio. A value of `50` means try to decode the file `50x` faster than real time, if possible.
+#               Use this setting to lower CPU peaks when computing replaygain tags.
+# @param file_name File name.
+def replaces file.replaygain(~id=null(), ~compute=true, ~ratio=50., file_name) =
   id = string.id.default(default="file.replaygain", id)
-  try
-    meta = file.metadata(exclude=["replaygain_track_gain"], fname)
-    replaygain_meta = meta["replaygain_track_gain"]
-    match = r/([\+\-\d\.]+)\s*dB/i.exec(replaygain_meta)
-    gain = float_of_string(list.assoc(1, match))
-    log.important(
-      label=id,
-      "Detected track replaygain #{gain} dB for #{string.quote(fname)}"
-    )
-
-    gain
-  catch _ do
-    r = request.create(resolve_metadata=false, fname)
-    if
-      request.resolve(r)
-    then
-      log.important(
-        label=id, "Computing replay gain for #{string.quote(fname)}"
-      )
+  file_name_quoted = string.quote(file_name)
 
-      t = time()
-      s = source.replaygain.compute(request.once(r))
-      source.drop(ratio=ratio, s)
-      gain = s.gain()
-      log.important(
-        label=id,
-        "Computed replay gain #{gain} dB for #{string.quote(
-          fname
-        )} (time: #{time() - t} s)."
-      )
+  _metadata = file.metadata(exclude=["replaygain_track_gain"], file_name)
+  gain = metadata.replaygain(_metadata)
 
-      gain
-    else
-      null()
+  if gain != null() then
+    log.important(label=id, "Detected track replaygain #{gain} dB for #{file_name_quoted}.")
```

Comment:
I would unescalate the log (eg `log.normal` or even `log.debug`), and same below